### PR TITLE
rspamd: Fixed build for macOS and improved Portfile

### PR DESCRIPTION
**Pull request by essandess**
* Build fixes segmentaton fault and -DCMAKE_OSX_ARCHITECTURES issues
* Add port:libstemmer dependency
* https://github.com/rspamd/rspamd/issues/2884
* https://github.com/rspamd/rspamd/issues/2883
* https://github.com/macports/macports-ports/commit/857c47b3beba97eb7e9934c5bff6e8982bf44307
* https://trac.macports.org/ticket/58437

###### Type(s)

- [x] bugfix
- [x] enhancement
- [x] security fix

###### Tested on

macOS 10.14.4 18E226
Xcode 10.2.1 10E1001 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: mail/rspamd/Portfile [hunk 1]
```diff
@@ -33,49 +30,80 @@ compiler.whitelist clang macports-clang-8.0 macports-clang-7.0 macports-clang-6.
 
 depends_build-append \
                     port:pkgconfig \
+                    port:ragel \
                     port:luajit \
-                    port:ragel
+                    port:gd2 \
+                    port:fann \
+                    port:libstemmer
```
Comment (cjones051073) [inline mail/rspamd/Portfile]: Just double checking, these 3 additions are really only build deps ? They aren’t needed at runtime ?

File: mail/rspamd/Portfile [hunk 2]
```diff
@@ -33,49 +30,80 @@ compiler.whitelist clang macports-clang-8.0 macports-clang-7.0 macports-clang-6.
...
+                    port:zlib \
+                    path:lib/pkgconfig/glib-2.0.pc:glib2 \
+                    path:lib/libopenblas.dylib:OpenBLAS
+
+# Critical dependency note: port:pcre and port:pcre2 break the rspamd binary;
+# use native /usr/lib/libpcre.dylib.  See https://github.com/rspamd/rspamd/issues/2884
+# Do not use this the library provided by port:pcre without pcre jit debugging
+# Do not use -DENABLE_PCRE2=ON without pcre jit debugging
+# # depends_build-append  port:pcre
+# # depends_build-append  port:pcre2
```
Comment (cjones051073) [inline mail/rspamd/Portfile]: Could you perhaps file a trac ticket against  MPs pcre port pointing at this issue. Just to document it incase anyone else stumbles on something similar or looks to addressing it ?

File: mail/rspamd/Portfile [hunk 3]
```diff
@@ -33,49 +30,80 @@ compiler.whitelist clang macports-clang-8.0 macports-clang-7.0 macports-clang-6.
...
+# Reference: https://github.com/rspamd/rspamd/blob/master/CMakeLists.txt
+# Also see https://github.com/rspamd/rspamd/issues/2884
 configure.args-append \
+    -DPCRE_ROOT_DIR=/usr/lib \
+    -DENABLE_LUAJIT=ON \
+    -DENABLE_TORCH=ON \
+    -DENABLE_SNOWBALL=ON \
+    -DENABLE_FANN=ON \
+    -DENABLE_GD=ON \
+    -DENABLE_LIBUNWIND=ON \
```
Comment (cjones051073) [inline mail/rspamd/Portfile]: This unwind option is causing build failures. See the azure build logs. e.g.

```
CMake Error at CMakeLists.txt:415 (MESSAGE):
  Cannot find library unwind for package LIBUNWIND
```

possibly you need to add `libunwind-headers` port as a build dep.

File: mail/rspamd/Portfile (hunk 2 above)
Comment (essandess) [reply mail/rspamd/Portfile]: https://trac.macports.org/ticket/58437

Comment (cjones051073) [reply mail/rspamd/Portfile]: Cheers

File: mail/rspamd/Portfile (hunk 1 above)
Comment (cjones051073) [reply mail/rspamd/Portfile]: Please don’t resolve a comment with actually saying or changing anything. If its ok then fine, but a comment to that affect would be good.

File: mail/rspamd/Portfile (hunk 3 above)
Comment (cjones051073) [reply mail/rspamd/Portfile]: Likewise, what have you changed here to resolve this ?

Comment (essandess) [reply mail/rspamd/Portfile]: Thanks. I added `port:unwind` under `depends_run`. Everything will build without, but the binary won't run without it.

File: mail/rspamd/Portfile (hunk 1 above)
Comment (essandess) [reply mail/rspamd/Portfile]: Thanks, corrected. See https://github.com/macports/macports-ports/pull/4258/files#r280977351

Comment (essandess) [reply mail/rspamd/Portfile]: Hold on. I see that there's a `libragel.dylib` too. `port:ragel` needs to go under `depends_lib`.

Also note that I added a `depends_run` section for ports wanted at runtime.

**Pull request merged**